client/src/pages/TaskHistoryPage/TaskHistoryPage.js

```diff
@@ -1,23 +1,71 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import styles from "./TaskHistoryPage.module.css";
 import TaskListView from "../../components/TaskListView/TaskListView.js";
+import {
+  buildDailyTaskRoute,
+  buildDateRangeRoute,
+  convertTaskData,
+  getWeekRange,
+  getMonthRange,
+} from "../../utils.js";
+import { ErrorMessage } from "../../components/ErrorMessage/ErrorMessage.js";
+import { instance } from "../../axios.js";
 
 const TaskHistoryPage = () => {
-  const sampleTask = {
-    title: "gdsc meeting",
-    location: "dh2020",
-    startTime: "05:00pm",
-    endTime: "07:00pm",
-    isOngoing: false,
+  const dayFilter = "day";
+  const weekFilter = "week";
+  const monthFilter = "month";
+  const allFilter = "all";
+  const [filter, setFilter] = useState(dayFilter);
+  const [taskData, setTaskData] = useState([]);
+  const [loadingError, setLoadingError] = useState(false);
+  const [loadingErrorMessage, setLoadingErrorMessage] = useState("");
+
+  const getTaskOnRoute = async (route) => {
+    setLoadingErrorMessage("");
+    await instance
+      .get(route)
+      .then((taskData) => {
```

Comment:
This should be labelled as `response` instead because when the API returns, the data will generally be found under `response.data`. Having `taskData.data` would seem a bit weird to have.

Reply:
Renamed accordingly